glue/plugins/wcs_autolinking/tests/test_wcs_autolinking.py

```diff
@@ -240,3 +240,129 @@ def test_celestial_with_unknown_axes():
     assert link[2].get_from_ids() == [px2, pz2]
     assert link[3].get_to_id() == py1
     assert link[3].get_from_ids() == [px2, pz2]
+
+
+def test_wcs_autolinking_of_2d_cube_with_temporal_and_spectral_axes_case_1():
+    """
+    A test to confirm that two 2D data cubes with matching number of dimensions
+    where the first is temporal and the next one spectral (vacuum wavelength in this
+    case) is indeed autolinked.
+    """
+
+    wcs1 = WCS(naxis=2)
+    wcs1.wcs.ctype = 'TIME', 'WAVE'
+    wcs1.wcs.set()
+
+    data1 = Data(label='Data 1')
+    data1.coords = wcs1
+    data1['x'] = np.ones((2, 3))
+    py1, px1 = data1.pixel_component_ids
+
+    wcs2 = WCS(naxis=2)
+    wcs2.wcs.ctype = 'TIME', 'WAVE'
+    wcs2.wcs.set()
+
+    data2 = Data(label='Data 2')
+    data2.coords = wcs2
+    data2['x'] = np.ones((2, 3))
+    py2, px2 = data2.pixel_component_ids
+
+    dc = DataCollection([data1, data2])
+    links = wcs_autolink(dc)
+    assert len(links) == 1
+    link = links[0]
+    assert isinstance(link, MultiLink)
+    assert len(link) == 4
+    assert link[0].get_to_id() == px2
+    assert link[0].get_from_ids() == [px1, py1]
+    assert link[1].get_to_id() == py2
+    assert link[1].get_from_ids() == [px1, py1]
+    assert link[2].get_to_id() == px1
+    assert link[2].get_from_ids() == [px2, py2]
+    assert link[3].get_to_id() == py1
+    assert link[3].get_from_ids() == [px2, py2]
+
+
+def test_wcs_autolinking_of_2d_cube_with_temporal_and_spectral_axes_case_2():
+    """
+    A test to confirm that two 2D data cubes with matching number of dimensions
+    where the one is spectral (air wavelength in this case) and the other one
+    temporal is indeed autolinked, to test that the order does not matter.
+    """
+
+    wcs1 = WCS(naxis=2)
+    wcs1.wcs.ctype = 'AWAV', 'TIME'
+    wcs1.wcs.set()
+
+    data1 = Data(label='Data 1')
+    data1.coords = wcs1
+    data1['x'] = np.ones((2, 3))
+    py1, px1 = data1.pixel_component_ids
+
+    wcs2 = WCS(naxis=2)
+    wcs2.wcs.ctype = 'TIME', 'AWAV'
+    wcs2.wcs.set()
+
+    data2 = Data(label='Data 2')
+    data2.coords = wcs2
+    data2['x'] = np.ones((2, 3))
+    py2, px2 = data2.pixel_component_ids
+
+    dc = DataCollection([data1, data2])
+    links = wcs_autolink(dc)
+    assert len(links) == 1
+    link = links[0]
+    assert isinstance(link, MultiLink)
+    assert len(link) == 4
+    assert link[0].get_to_id() == px2
+    assert link[0].get_from_ids() == [px1, py1]
+    assert link[1].get_to_id() == py2
+    assert link[1].get_from_ids() == [px1, py1]
+    assert link[2].get_to_id() == px1
+    assert link[2].get_from_ids() == [px2, py2]
+    assert link[3].get_to_id() == py1
+    assert link[3].get_from_ids() == [px2, py2]
+
+
+def test_has_celestial_with_time_and_spectral_axes():
```

Comment:
Added this use case for the wcs auto-linking between some IRIS Level 2 multiscan data sets with 4D sequentially-stacked data cubes and 3D SJI data cubes to be used in demo docs.